### ISSUE TITLE
Why the loss is "nan"

### ISSUE THREAD

**Issue by congmingyige** [WZH0120/SAM2-UNet#43]
<img width="260" alt="Image" src="https://github.com/user-attachments/assets/3e1e27ef-ea4e-41f6-84a7-1c4667a7c105" />

Data: 160x160 size, 75000 images

It is ok for running 20 epochs in the first time. Then I changes dataset and run for the second time, it meets mistake of "loss nan", and it happens suddenly, the previous losses have no problem.

**Comment by xiongxyowo** [comment]
Hi, you can try to modify the Adapter code as follows to stabilize the training process:
```
from timm.models.layers import trunc_normal_
class Adapter(nn.Module):
    def __init__(self, blk) -> None:
        super(Adapter, self).__init__()
        self.block = blk
        dim = blk.attn.qkv.in_features
        self.prompt_learn = nn.Sequential(
            nn.Linear(dim, 32),
            nn.GELU(),
            nn.Linear(32, dim),
            nn.GELU()
        )
        self.init_weights()

    def forward(self, x):
        prompt = self.prompt_learn(x)
        promped = x + prompt
        net = self.block(promped)
        return net
    
    def init_weights(self):
        def _init_weights(m):
            if isinstance(m, nn.Linear):
                print(m)
                trunc_normal_(m.weight, std=.02)
                if isinstance(m, nn.Linear) and m.bias is not None:
                    nn.init.constant_(m.bias, 0)
            elif isinstance(m, nn.LayerNorm):
                nn.init.constant_(m.bias, 0)
                nn.init.constant_(m.weight, 1.0)
        self.prompt_learn.apply(_init_weights)
```

**Comment by congmingyige** [comment]
Thanks. The "nan" problem does not happen in the newest training.
